src/pages/dashboard/index.jsx

```diff
@@ -1,83 +1,120 @@
+  const [isStudentModalVisible, setIsStudentModalVisible] = useState(false);
+  const [selectedProfileId, setSelectedProfileId] = useState(null);
+  const [teachers, setTeachers] = useState([]);
+  const [students, setStudents] = useState([]);
+  const [isTeacher, setIsTeacher] = useState(false);
+  const menuRef = useRef(null);
+  const navigate = useNavigate();
+  const { currentUser } = useUser();
 
   useEffect(() => {
-    getUsers().then(setCohorts)
-  }, [])
+    getUsers().then(setCohorts);
+    getTeachers();
+    getStudents();
+    verifyTeacher()
+  }, []);
 
   const onClickSearchBar = () => {
-    setIsSearchResVisible(true)
-  }
+    setIsSearchResVisible(true);
+  };
 
   const onChange = (e) => {
-    setSearchVal(e.target.value)
-  }
-
-  // console.log(cohorts);
+    setSearchVal(e.target.value);
+  };
 
   const result = cohorts.filter((cohort) => {
     if (cohort.firstName && cohort.lastName) {
-      const fullName = `${cohort.firstName || ''} ${
-        cohort.lastName || ''
-      }`.toLowerCase()
-      return fullName.includes(searchVal.toLowerCase())
+      const fullName = `${cohort.firstName || ""} ${
+        cohort.lastName || ""
+      }`.toLowerCase();
+      return fullName.includes(searchVal.toLowerCase());
     }
-  })
+  });
 
   const handleClickOutside = (event) => {
     if (
-      !event.target.closest('.search-cohorts-results') &&
-      !event.target.closest('#input-wrapper-search-bar')
+      !event.target.closest(".search-cohorts-results") &&
+      !event.target.closest("#input-wrapper-search-bar")
     ) {
-      setIsSearchResVisible(false)
-      setIsStudentModalVisible(false)
-      setSelectedProfileId(null)
+      setIsSearchResVisible(false);
+      setIsStudentModalVisible(false);
+      setSelectedProfileId(null);
     }
-  }
+  };
 
   useEffect(() => {
-    document.addEventListener('click', handleClickOutside)
+    document.addEventListener("click", handleClickOutside);
     return () => {
-      document.removeEventListener('click', handleClickOutside)
-    }
-  }, [])
+      document.removeEventListener("click", handleClickOutside);
+    };
+  }, []);
 
   const onClickStudent = (id) => {
-    setSelectedProfileId(id)
-    setIsStudentModalVisible(true)
-  }
+    setSelectedProfileId(id);
+    setIsStudentModalVisible(true);
+  };
 
-  const { openModal, setModal } = useModal()
+  const { openModal, setModal } = useModal();
 
   const showModal = () => {
-    setModal('Create a post', <CreatePostModal />)
-    openModal()
-  }
+    setModal("Create a post", <CreatePostModal />);
+    openModal();
+  };
+
+  const allSearchResults = () =>
+    navigate("/search-results", { state: { results: result, searchVal } });
+
+  const getTeachers = () => {
+    if (cohorts.length > 1) {
+      const findTeacher = cohorts.map((u) => {
+        if (u.role === "TEACHER") return u;
+      });
+      const isTeacher = findTeacher.filter((u) => {
+        if (u !== undefined) return u;
+      });
```

Comment:
filter will accept a function that returns a boolean to determine if items should be in the result, so this could be shortened to:

```js
const isTeacher = findTeacher.filter(u => u !== undefined)
```